src/middlewared/middlewared/plugins/disk_/disk_info.py

```diff
@@ -7,6 +8,31 @@
 from middlewared.service import CallError, private, Service
 
 
+def get_partition_size_info(disk_name, s_offset, s_size):
+    """Kernel sysfs reports most disk files related to "size" in 512 bytes.
+    To properly calculate the starting SECTOR of partitions, you must
+    look at logical_block_size (again, reported in 512 bytes) and
+    do some calculations. It is _very_ important to do this properly
+    since almost all userspace tools that format disks expect partition
+    positions to be in sectors."""
+    lbs = start_sect = end_sect = sect_size = 0
+    with contextlib.suppress(FileNotFoundError, ValueError):
+        with open(f'/sys/block/{disk_name}/queue/logical_block_size') as f:
+            lbs = int(f.read().strip())
+
+    if not lbs:
+        # this should never happen
```

Comment:
Do we need to raise an exception or log an error message?

Reply:
I'm afraid to since this is landing in 24.04.2 and this function is being called in critical paths. I'd rather just return without crashing since I'm not sure what other regressions may occur if I raise an exception. I'll push a more involved fix for EE+ to clean all of this up.